psycop/common/sequence_models/tasks.py

```diff
@@ -150,3 +153,141 @@ def configure_optimizers(
             **self.scheduler_kwargs,
         )
         return [optimizer], [lr_scheduler]
+
+
+class EncoderForClassification(pl.LightningModule):
+    """
+    A BEHRT model for the classification task.
+    """
+
+    def __init__(
+        self,
+        embedding_module: BEHRTEmbedder,
+        encoder_module: nn.Module,
+        aggregation_module: AggregationModule,
+        optimizer_kwargs: dict[str, Any],
+        lr_scheduler_kwargs: dict[str, Any],
+        num_classes: int = 1,
+    ):
+        super().__init__()
+        self.embedding_module = embedding_module
+        self.encoder_module = encoder_module
+        self.optimizer_kwargs = optimizer_kwargs
+        self.scheduler_kwargs = lr_scheduler_kwargs
+        self.aggregation_module = aggregation_module
+
+        self.d_model: int = embedding_module.d_model
+
+        self.is_binary = num_classes == 1
+        self.num_classes = num_classes
+        if self.is_binary:
+            self.loss = nn.BCEWithLogitsLoss()
+        else:
+            self.loss = nn.CrossEntropyLoss()
+
+        self.classification_head = nn.Linear(self.d_model, num_classes)
+
+        self.metrics = self.create_metrics(num_classes)
+
+    @staticmethod
+    def create_metrics(num_classes: int) -> dict[str, Metric]:
+        is_binary = num_classes == 1
+        if is_binary:
+            return {"AUROC": BinaryAUROC()}
+        return {"AUROC (macro)": MulticlassAUROC(num_classes=num_classes)}
+
+    def training_step(self, batch: BatchWithLabels, batch_idx: int) -> torch.Tensor:  # type: ignore # noqa: ARG002
+        output = self.forward(batch[0], batch[1])
+        loss = output["loss"]
+        self.log_step("Training", output)
+        return loss
+
+    def validation_step(self, batch: BatchWithLabels, batch_idx: int) -> torch.Tensor:  # type: ignore  # noqa: ARG002
+        output = self.forward(inputs=batch[0], labels=batch[1])
+        self.log_step("Validation", output)
+        return output["loss"]
+
+    def log_step(
+        self,
+        mode: Literal["Validation", "Training"],
+        output: dict[str, torch.Tensor],
+    ) -> None:
+        """
+        Logs the metrics for the given mode.
+        """
+        for metric_name, metric in output.items():
+            self.log(f"{mode} {metric_name}", metric)
+
+    def forward(  # type: ignore
```

Comment:
```suggestion
    def forward(  # type: ignore
```

Why is the type: ignore needed here? :-)